sdk/include/opentelemetry/sdk/trace/recordable.h

```diff
@@ -33,6 +34,14 @@ class Recordable
                       opentelemetry::trace::SpanId span_id,
                       opentelemetry::trace::SpanId parent_span_id) noexcept = 0;
 
+  /**
+   * Set an attribute of a span.
+   * @param name the name of the attribute
+   * @param value the attribute value
+   */
+  virtual void SetAttribute(nostd::string_view key,
+                            const opentelemetry::common::AttributeValue &&value) noexcept = 0;
```

Comment:
This is the wrong way to do move semantics. 

It should be either `const opentelemetry::common::AttributeValue &value` (i.e. doesn't allow move semantics) or `opentelemetry::common::AttributeValue && value` (supports move semantics).

Move semantics require modifying value so `const opentelemetry::common::AttributeValue &&` makes no sense.

Reply:
Thanks for chiming in @rnburn. I submitted PR #141 to fix this.